Bikes/Bikes.Application/Services/AnalyticsService.cs

```diff
@@ -0,0 +1,137 @@
+﻿using Bikes.Domain.Models;
+using Bikes.Domain.Repositories;
+
+namespace Bikes.Application.Services;
+
+/// <summary>
+/// A class that implements the interface of the AnalyticsService class
+/// </summary>
+public class AnalyticsService : IAnalyticsService
+{
+    private readonly IRepository<Bike, int> _bikeRepository;
+    private readonly IRepository<BikeModel, int> _bikeModelRepository;
+    private readonly IRepository<Rent, int> _rentRepository;
+    private readonly IRepository<Renter, int> _renterRepository;
+
+    /// <summary>
+    /// The constructor that initializes repositories
+    /// </summary>
+    /// <param name="bikeRepository"></param>
+    /// <param name="bikeModelRepository"></param>
+    /// <param name="rentRepository"></param>
+    /// <param name="renterRepository"></param>
+    public AnalyticsService(
+        IRepository<Bike, int> bikeRepository,
+        IRepository<BikeModel, int> bikeModelRepository,
+        IRepository<Rent, int> rentRepository,
+        IRepository<Renter, int> renterRepository)
+    {
+        _bikeRepository = bikeRepository;
+        _bikeModelRepository = bikeModelRepository;
+        _rentRepository = rentRepository;
+        _renterRepository = renterRepository;
+    }
+
+    /// <summary>
+    /// A method that returns information about all sports bikes
+    /// </summary>
+    public List<Bike> GetSportBikes()
+    {
+        return _bikeRepository.ReadAll()
+            .Where(bike => bike.Model.Type == BikeType.Sport)
+            .ToList();
```

Comment:
Упростить инициализацию коллекций. Тут и далее по коду.

```suggestion
        return [.. bikeRepository.ReadAll().Where(bike => bike.Model.Type == BikeType.Sport)];
```